### PR TITLE
Docc docs

### PR DESCRIPTION
**Pull request by aboedo**
Added lanes to automatically deploy Docc-format docs. 

The docs are available here: https://revenuecat-docs.netlify.app/documentation/Revenuecat

Additionally, I added 2 lanes: 
- `compile-docs`: generates the docs and stores them in `LatestTagDocs` (I don't love that name, suggestions welcome). This is incorporated into the `bump` lane as well, so we don't forget to do it. 
- `deploy-docs`: currently just calls a webhook that deploys the docs automatically. This is called automatically when a tag is pushed (or it will be, once we re-enable it in circleCI, after 4.0.0 is stable).

I removed `jazzy` from the fastfile and circleci config as well. 

There are a couple of improvements still to do to the docs, but in general terms, this should be workable. 

To-do after this is merged: 
- [ ] Update Netlify config so it points to `main` branch (it's currently pointing to this branch)
- [ ] Transfer Netlify app to company account
- [ ] Add DNS setup to Netlify app

### CODE REVIEW

File: fastlane/Fastfile [hunk 1]
```diff
@@ -259,6 +260,37 @@ platform :ios do
     )
   end
 
+  desc "Compiles Docc archive"
```
Comment (aboedo) [inline fastlane/Fastfile]: this is probably the interesting bit to review

File: fastlane/Fastfile [hunk 2]
```diff
@@ -259,6 +260,37 @@ platform :ios do
...
+  lane :compile_docs do
+    Dir.chdir("..") do
+      Dir.mktmpdir {|dir|
+        archive_name = 'RevenueCat.doccarchive'
+        
+        sh('xcodebuild', 'docbuild', '-scheme', 'RevenueCat', '-derivedDataPath', dir, '-destination', 'platform=iOS Simulator,name=iPhone 12')
+        
+        file_path = Dir["#{dir}/**/#{archive_name}"].first
+        FileUtils.copy_entry(file_path, "LatestTagDocs/#{archive_name}")
```
[the remark this answers is not in the document]
Comment (aboedo) [reply fastlane/Fastfile]: Also, thoughts on this being a folder in this repo? It seemed to me like a very simple way to deploy this, and I kinda like that it's easy for us to see whenever needed

File: LatestTagDocs/RevenueCat.doccarchive/_redirects [hunk 3]
```diff
@@ -0,0 +1 @@
+/* /index.html 200
```
Comment (aboedo) [inline LatestTagDocs/RevenueCat.doccarchive/_redirects:1]: this one is required for deploying Docc. 

Also, files from here on are auto-generated, other than the Fastfile at the very bottom of this diff

File: .circleci/config.yml [hunk 4]
```diff
@@ -235,21 +235,9 @@ jobs:
...
-      - run:
-          name: Install awscli
-          command: sudo python3 -m pip install awscli
-      - run:
-          name: Deploy to S3
-          command: aws s3 sync generated_docs s3://purchases-docs/ios --delete
-      - run:
-          name: Invalidate CloudFront caches
-          command: aws cloudfront create-invalidation --distribution-id EPTW7F3CB566V --paths "/*"
+          command: bundle exec fastlane deploy_docs
```
Comment (aboedo) [inline .circleci/config.yml:240]: so much simpler

**Review by taquitos — APPROVED** [review]
📚 
globglogabgalab

**Pull request merged**